src/Rating.js

```diff
@@ -1,8 +1,228 @@
 'use strict';
 
 import RatingBase from './Rating.soy';
+import { core } from 'metal';
 
 class Rating extends RatingBase {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', 'a', this.handleMouseOverEvent.bind(this));
+        this.delegate('click', 'a', this.handleClickEvent.bind(this));
+        this.on('mouseout', this.handleMouseOutEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.target.dataset.index, 10);
+
+            if (this.selectedIndex === index && this.canReset) {
+                this.reset();
+            }
+            else {
+                this.selectedIndex = index;
+            }
+        }
+    }
+
+    /**
+     * Reset rating attributes to its initial value
+     * @protected
+     */
+    reset() {
+        this.rate = null;
+        this.selectedIndex = -1;
+    }
+
+    /**
+     * Handles mouseout event
+     * @protected
+     */
+    handleMouseOutEvent() {
+        this.currentMouseTarget = undefined;
```

Comment:
It's a convention, but we're using `_` as a suffix instead of a prefix on metal, as [google](https://github.com/google/closure-library/blob/master/closure/goog/base.js#L117) uses it. Eduardo prefers it like this.